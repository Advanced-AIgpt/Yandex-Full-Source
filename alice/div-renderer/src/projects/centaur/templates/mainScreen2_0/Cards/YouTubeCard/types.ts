import { IAbstractCardProps } from '../types';

export interface IYouTubeProps extends IAbstractCardProps {
    type: 'youtube';
}
