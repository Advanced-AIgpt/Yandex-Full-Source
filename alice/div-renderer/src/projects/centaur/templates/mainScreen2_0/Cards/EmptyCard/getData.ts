import { IEmptyCardProps } from './types';
import { ICardDataAdapter } from '../types';

export const getEmptyCardData: ICardDataAdapter<IEmptyCardProps> = function getEmptyCardData(card, requestState) {
    if (card && card.VacantCardData) {
        return {
            type: 'empty',
            requestState,
        };
    }
    return null;
};
