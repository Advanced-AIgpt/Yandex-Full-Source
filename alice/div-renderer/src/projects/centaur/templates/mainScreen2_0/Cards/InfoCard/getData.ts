import { sample } from 'lodash';
import { IInfoCardProps } from './types';
import { DISCOVERY_CARDS } from '../../../mainScreen/discovery';
import { ICardDataAdapter } from '../types';
import { textAction } from '../../../../../../common/actions';

export const getInfoCardData: ICardDataAdapter<IInfoCardProps> = function getInfoCardData(card, requestState) {
    if (typeof card.InfoCardData !== 'undefined' && card.InfoCardData !== null) {
        const data = card.InfoCardData.Title === 'test' ? DISCOVERY_CARDS[0] : sample(DISCOVERY_CARDS) || DISCOVERY_CARDS[0];

        return {
            type: 'info',
            title: data.title,
            image_background: data.bg,
            description: data.description,
            requestState,
            actions: [
                {
                    log_id: 'discovery.mainscreen.execute',
                    url: textAction(data.description),
                },
            ],
        };
    }
    return null;
};
