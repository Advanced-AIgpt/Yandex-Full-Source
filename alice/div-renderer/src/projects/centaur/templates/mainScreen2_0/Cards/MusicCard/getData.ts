import { IMusicCardProps } from './types';
import { colorDarkGrey } from '../../../../style/constants';
import { ICardDataAdapter } from '../types';

const playlists: readonly [string, string][] = [
    ['Плейлист дня', '#31AD54'],
    ['Мне нравится', '#E53866'],
    ['Тайник', '#186FD3'],
    ['Подкасты', '#036292'],
    ['Дежавю', '#A019D1'],
    ['Премьера', '#D1673E'],
    ['Плейлист с Алисой', '#2D0B87'],
    ['Мой 2021', '#C20146'],
    ['Плейлист семьи', '#FE8853'],
];

export const getMusicCardData: ICardDataAdapter<IMusicCardProps> = function getMusicCardData(card, requestState) {
    if (typeof card.MusicCardData !== 'undefined' && card.MusicCardData !== null) {
        const data = card.MusicCardData;

        const knownPlaylist = playlists.find(el => el[0] === data.Name);

        const color = knownPlaylist ? knownPlaylist[1] : colorDarkGrey;

        return {
            type: 'music',
            name: data.Name,
            modified: data.Modified?.value,
            color,
            cover: data.Cover,
            actions: (card.Action && [
                {
                    log_id: 'music_card_action',
                    url: card.Action,
                },
            ]) || undefined,
            preventDefaultLoader: true,
            requestState,
        };
    }
    return null;
};
