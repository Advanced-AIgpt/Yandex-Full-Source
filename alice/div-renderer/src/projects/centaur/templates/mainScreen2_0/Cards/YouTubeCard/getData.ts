import { compact } from 'lodash';
import { IYouTubeProps } from './types';
import { EnumLayer } from '../../../../actions/client';
import { ICardDataAdapter } from '../types';

export const getYouTubeCardData: ICardDataAdapter<IYouTubeProps> = function getYouTubeCardData(card, requestState) {
    if (typeof card.YouTubeCardData !== 'undefined' && card.YouTubeCardData !== null) {
        return {
            type: 'youtube',
            actions: compact([
                card.Action && {
                    log_id: 'main_screen_youtube_action',
                    url: card.Action,
                },
            ]),
            layer: EnumLayer.content,
            requestState,
        };
    }
    return null;
};
