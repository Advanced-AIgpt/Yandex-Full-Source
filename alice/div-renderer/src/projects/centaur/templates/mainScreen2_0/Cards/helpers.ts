export function getCardMainScreenId({ rowIndex, colIndex }: { rowIndex?: number; colIndex?: number }) {
    return `main_screen_card_${rowIndex || 0}_${colIndex || 0}`;
}
