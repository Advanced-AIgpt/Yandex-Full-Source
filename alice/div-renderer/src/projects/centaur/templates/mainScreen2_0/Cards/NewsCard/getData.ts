import { compact } from 'lodash';
import { INewsCardProps } from './types';
import { EnumLayer } from '../../../../actions/client';
import { getImageByTopic } from '../../../../helpers/newsHelper';
import { ICardDataAdapter } from '../types';

export const getNewsCardData: ICardDataAdapter<INewsCardProps> = function getNewsCardData(card, requestState) {
    if (typeof card.NewsCardData !== 'undefined' && card.NewsCardData !== null) {
        const data = card.NewsCardData;

        return {
            type: 'news',
            title: data.Title,
            content: data.Content,
            image: data.ImageUrl || getImageByTopic(data.Topic) || undefined,
            layer: EnumLayer.dialog,
            requestState,
            actions: compact([
                card.Action && {
                    log_id: 'news_card_action',
                    url: card.Action,
                },
            ]),
        };
    }
    return null;
};
