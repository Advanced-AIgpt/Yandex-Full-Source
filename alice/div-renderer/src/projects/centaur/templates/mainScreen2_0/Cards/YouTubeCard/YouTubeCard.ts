import { FixedSize, ImageBlock, MatchParentSize, TextBlock } from 'divcard2';
import { BasicCard } from '../BasicCard';
import { title32m } from '../../../../style/Text/Text';
import { colorWhiteOpacity50 } from '../../../../style/constants';
import EmptyDiv from '../../../../components/EmptyDiv';
import { getS3Asset } from '../../../../helpers/assets';
import { IYouTubeProps } from './types';
import { getCardMainScreenId } from '../helpers';

export default function YouTubeCard({ actions, longtap_actions, rowIndex, colIndex } : IYouTubeProps) {
    const cardData: Parameters<typeof BasicCard>[0] = {
        id: getCardMainScreenId({ rowIndex: rowIndex || 0, colIndex: colIndex || 0 }),
        items: [
            new TextBlock({
                ...title32m,
                text_color: colorWhiteOpacity50,
                text: 'Youtube',
            }),
            new EmptyDiv({
                height: new MatchParentSize({ weight: 31 }),
            }),
            new ImageBlock({
                alignment_horizontal: 'center',
                alignment_vertical: 'center',
                height: new FixedSize({ value: 112 }),
                width: new FixedSize({ value: 161 }),
                image_url: getS3Asset('main-screen/cards/youtube/logo.png'),
                margins: {
                    top: 31,
                },
            }),
            new EmptyDiv({
                height: new MatchParentSize({ weight: 56 }),
            }),
        ],
        actions,
        longtap_actions,
    };

    return BasicCard(cardData);
}
