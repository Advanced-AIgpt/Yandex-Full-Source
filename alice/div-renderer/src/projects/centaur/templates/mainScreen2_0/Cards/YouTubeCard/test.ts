import YouTubeCard from './YouTubeCard';
import { AnonymizeDataForSnapshot } from '../../../../../../common/helpers/dev';
import { logger } from '../../../../../../common/logger';
import { NAlice } from '../../../../../../protos';
import { getYouTubeCardData } from './getData';
const TCardData = NAlice.NData.TCentaurMainScreenMyScreenData.TColumn.TCardData;

jest.mock('../../../../../../common/logger');

describe('Main screen empty card', () => {
    it('should get actual data', () => {
        expect(getYouTubeCardData(new TCardData({}))).toBeNull();

        expect(getYouTubeCardData(new TCardData({
            NewsCardData: {},
        }))).toBeNull();

        expect(getYouTubeCardData(new TCardData({
            YouTubeCardData: {},
        }))).not.toBeNull();
    });

    it('should match snapshot', () => {
        expect(AnonymizeDataForSnapshot(YouTubeCard({
            type: 'youtube',
        }))).toMatchSnapshot();

        expect(logger.error).toHaveBeenCalledTimes(0);
    });
});
