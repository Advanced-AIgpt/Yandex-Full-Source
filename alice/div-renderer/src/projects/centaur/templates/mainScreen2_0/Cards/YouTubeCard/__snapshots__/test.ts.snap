// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main screen empty card should match snapshot 1`] = `
Object {
  "background": Array [
    Object {
      "color": "#1aeaf6ec",
      "type": "solid",
    },
  ],
  "border": Object {
    "corner_radius": 28,
  },
  "height": Object {
    "type": "match_parent",
    "weight": 1,
  },
  "id": "main_screen_card_0_0",
  "items": Array [
    Object {
      "font_size": 32,
      "font_weight": "medium",
      "line_height": 40,
      "text": "Youtube",
      "text_color": "#80ffffff",
      "type": "text",
    },
    Object {
      "delimiter_style": Object {
        "color": "#0fff",
      },
      "height": Object {
        "type": "match_parent",
        "weight": 31,
      },
      "type": "separator",
    },
    Object {
      "alignment_horizontal": "center",
      "alignment_vertical": "center",
      "height": Object {
        "type": "fixed",
        "value": 112,
      },
      "image_url": "https://dialogs.s3.yandex.net/smart_displays/main-screen/cards/youtube/logo.png",
      "margins": Object {
        "top": 31,
      },
      "type": "image",
      "width": Object {
        "type": "fixed",
        "value": 161,
      },
    },
    Object {
      "delimiter_style": Object {
        "color": "#0fff",
      },
      "height": Object {
        "type": "match_parent",
        "weight": 56,
      },
      "type": "separator",
    },
  ],
  "paddings": Object {
    "bottom": 24,
    "left": 24,
    "right": 24,
    "top": 24,
  },
  "type": "container",
  "width": Object {
    "type": "match_parent",
    "weight": 1,
  },
}
`;
